### DISCUSSION TITLE
[Pytest] asyncTest with no current event loop main thread

### DISCUSSION THREAD

**Discussion by wjs2063** [fastapi/fastapi#9558] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
Here is my test_async.py Code

`
import pytest
from httpx import AsyncClient
from main import app
from models.database import *
import asyncio
"""
This is a async Test File 
"""




base_url = "http://111.111.1.1:54321"


async def override_async_db():
    db = motor_asyncio.AsyncIOMotorClient(LOCAL_TESTDB)
    db.get_io_loop = asyncio.get_event_loop
    try :
        yield db.local
    finally :
        db.close()

app.dependency_overrides[asyncdb] = override_async_db


@pytest.mark.anyio
async def test_create_user():
    async with AsyncClient(app = app,base_url = base_url)  as ac:
        response = await ac.post("/api/user/async",
    json = {"name": "jaehyeon",
       "user_id": "aaa1234",
       "email": "kkk@naver.com",
       "password": "1234567",
       "nickname": "user1",
       "disabled": "False"
    }
    )
    assert response.status_code == 201
````

main.py is here

````
from fastapi import FastAPI
from routes.route import apps

app = FastAPI()
app.include_router(apps)

````

Here is route.py for example
````
@apps.post("/api/user/async",response_model = UserData,status_code = 201)
async def create_user_async(user :User,db: Annotated[motor_asyncio.AsyncIOMotorClient ,Depends(asyncdb)]):
    user = dict(user)
    await db.users.insert_one(user)
    data = dict(await db.users.find_one(user))
    # serialize
    data["_id"] = str(data["_id"])
    return data
````
```


### Description

Error message 
````

     self = <asyncio.unix_events._UnixDefaultEventLoopPolicy object at 0xffff8aff0050>

     def get_event_loop(self):
     Get the event loop for the current context.
    
        Returns an instance of EventLoop or raises an exception.
        if (self._local._loop is None and
                not self._local._set_called and
                threading.current_thread() is threading.main_thread()):
            self.set_event_loop(self.new_event_loop())
    
        if self._local._loop is None:
            raise RuntimeError(There is no current event loop in thread %r.
                               % threading.current_thread().name)
         RuntimeError: There is no current event loop in thread 'MainThread

      /usr/lib/python3.11/asyncio/events.py:677: RuntimeError

      FAILED test_async.py::test_create_user[trio] - RuntimeError: There is no current event loop in thread 'MainThread'.
````

What i miss it ? 


### Operating System

macOS

### Operating System Details

Docker : alpine 3.8
Dekstop OS : m1 mac 

python version 3.11.1
fastapi 0.95 



### FastAPI Version

0.95

### Python Version

3.11.1

### Additional Context

In my case 

pytest result is always error .

client.post is just insert docs to mongodb .

**Comment by wjs2063** [comment, marked as answer]
I solved it 

pytest-asyncio 
just change @pyest.mark.asyncio is answer

**Comment by houmie** [comment]
@wjs2063 I have the same problem. How did you solve it? I don't understand your answer here. I already have ```@pytest.mark.asyncio``` on top of my unit test.

**Comment by tcztzy** [comment]
How to actually fix this problem with `anyio` & `trio`?
